lib/segment/src/types.rs

```diff
@@ -707,6 +707,14 @@ pub struct StrictModeConfig {
     /// Max batchsize when upserting
     #[serde(skip_serializing_if = "Option::is_none")]
     pub upsert_max_batchsize: Option<usize>,
+
+    /// Max number of read operations per second
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub read_rate_limit_per_sec: Option<usize>,
+
+    /// Max number of write operations per second
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub write_rate_limit_per_sec: Option<usize>,
```

Comment:
Not sure about this one, but I think it may be useful to be a bit more explicit on this one:

```suggestion
    /// Max number of read operations per second per shard per peer
    #[serde(skip_serializing_if = "Option::is_none")]
    pub read_rate_limit_per_sec: Option<usize>,

    /// Max number of write operations per second per shard per peer
    #[serde(skip_serializing_if = "Option::is_none")]
    pub write_rate_limit_per_sec: Option<usize>,
```

Reply:
On another thought `per replica` may be shorter to `per shard per peer`